website/auth.py

```diff
@@ -151,11 +151,11 @@ def login():
         body = request.json
         # Validations
         if not isinstance(body, dict):
-            return 'body must be an object', 400
+            return 'not_object', 400
```

Comment:
Also, I would probably do some structured data like `{ code: 'not_object', message: 'body must be an object' }`.

Reply:
Didn't think about it that way! As the translations are already available to the template at the front-end I thought this would be the most straight-forward implementation. I agree that it would be better to return the translated error message from the back-end, but we would have to include the translated files to `auth.py`

Reply:
We should be sticking the translations on the global `g` object, so they're straight up available in the language we need, like this:

https://github.com/Felienne/hedy/blob/34aec748748cb1c30be932b04a26f415f00656f1/app.py#L280

It's a bit unfortunate the "auth" messages have their own section, so now we need to do hulahoops to get the information in the right place, but so be it...

I can see how you might just want to merge what you have today though, so you can also disregard what I'm saying and merge what you have. It's already better than what it used to be.

Reply:
Just re-wrote all translations to be created on the back-end. In the case of no response text given we still get the `ajax_error` in the front-end as fallback.